### ISSUE TITLE
Hang when copying files to Fairphone 5 device

### ISSUE THREAD

**Issue by ssssam** [libmtp/libmtp#227]
Hello,

I have an issue copying files to Fairphone 5 device on Linux, where the copy creates a zero byte file on the device, then hangs for a while and exits with an IO error.

On Microsoft Windows the same file copies successfully onto the device, so I'm working on the assumption that it's an issue in our MTP stack somewhere.

I will need more time to continue debugging, but here's a log of debug output from libmtp when I use [simple-mtpfs](https://github.com/phatina/simple-mtpfs) to do the copy. (The same hang occurs when using Nautilus + GVFS, but its harder to get debug output there).

cp ~/Podcasts/EP.61\ -\ ROMESH\ RANGANATHAN.mp3 ~/mnt

```
unique: 340, opcode: LOOKUP (1), nodeid: 1, insize: 71, pid: 4990
LOOKUP /EP.61 - ROMESH RANGANATHAN.mp3
getattr /EP.61 - ROMESH RANGANATHAN.mp3
   unique: 340, error: -2 (No such file or directory), outsize: 16
unique: 342, opcode: LOOKUP (1), nodeid: 1, insize: 71, pid: 4990
LOOKUP /EP.61 - ROMESH RANGANATHAN.mp3
getattr /EP.61 - ROMESH RANGANATHAN.mp3
   unique: 342, error: -2 (No such file or directory), outsize: 16
unique: 344, opcode: CREATE (35), nodeid: 1, insize: 87, pid: 4990
create flags: 0x80c1 /EP.61 - ROMESH RANGANATHAN.mp3 0100644 umask=0022
   create[9] flags: 0x80c1 /EP.61 - ROMESH RANGANATHAN.mp3
getattr /EP.61 - ROMESH RANGANATHAN.mp3
   NODEID: 3
   unique: 344, success, outsize: 160
unique: 346, opcode: GETXATTR (22), nodeid: 3, insize: 68, pid: 4990
   unique: 346, error: -38 (Function not implemented), outsize: 16
unique: 348, opcode: WRITE (16), nodeid: 3, insize: 4176, pid: 4990
write[9] 4096 bytes to 0 flags: 0x8001
```

^^^ ...now it hangs for a while ... ^^^

```
   unique: 348, error: -5 (Input/output error), outsize: 16
unique: 350, opcode: FLUSH (25), nodeid: 3, insize: 64, pid: 4990
flush[9]
   unique: 350, success, outsize: 16
unique: 352, opcode: RELEASE (18), nodeid: 3, insize: 64, pid: 0
release[9] flags: 0x8001
```

It's possible that the GETXATTR call is the issue. It's also possible that the error is on the device side after receiving an unexpected function call -- will try to raise with the  manufacturer as well.

**Comment by ssssam** [comment]
This seems to be have been an issue with the device itself.

For posterity, the workaround was this:

> If you are trying to copy your data from the phone to a computer via USB - selecting "Transfer Files (MTP)" - but some or all of your files are not visible, follow these steps:
>
>  1. Go to Settings > Apps & notifications > See all apps.
>  2. Tap on the three dots on the top right corner and select Show system.
>  3. Select External Storage > Storage > Clear Cache and Clear Storage.
>  4. Go back to the list with all the apps and select Media Storage > Storage > Clear Cache and Clear Storage.
>  5. Reboot the device.
>
> After doing this, you should be able to see the data on your phone.

**Comment by ssssam** [comment]
Update on this issue, for posterity...

  * I wrote to Fairphone support, who didn't have much idea what could be wrong. They did say that file copies are known to work on Ubuntu, while my issue is with Fedora.
  * Testing with `mtp-connect --sendfile myfile.mp3,/Podcasts` mostly works, so I can use this to transfer files.
  * I found an issue where a file with a quote mark in its name triggers an IO error -- maybe that's related to the issues I had above.
